src/lambda/wizards/samDeployWizard.ts

```diff
@@ -413,10 +426,45 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
         const val = picker.verifySinglePickerOutput(choices)
 
         return val?.label && ![loadingBuckets, messages.noBuckets, messages.bucketError].includes(val.label)
-            ? val.label
-            : undefined
+        ? val.label
+        : undefined
     }
 
+    public async promptForNewS3BucketName(step: number, selectedRegion: string): Promise<string | undefined>
+    {
+        this.additionalSteps++
+        const inputBox = input.createInputBox({
+            buttons: [this.helpButton, vscode.QuickInputButtons.Back],
+            options: {
+                title: localize('AWS.s3.createBucket.prompt', 'Enter a new bucket name'),
```

Comment:
We discussd re-using `createBucketCommand` (and make its `node` parameter optional). I guess  the need for `this.additionalSteps--` etc. makes that awkward?

 But, this is missing the telemetry from `createBucketCommand`, at least.

Reply:
Is there information that needs to be pulled from the node? re: above also, you can just add an optional parameter indicating step numbers if necessary.

Reply:
+1 on telemetry